server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2441,9 +2412,7 @@ protected void commitIndexWriter(final IndexWriter writer, final Translog transl
                 commitData.put(SequenceNumbers.MAX_SEQ_NO, Long.toString(localCheckpointTracker.getMaxSeqNo()));
                 commitData.put(MAX_UNSAFE_AUTO_ID_TIMESTAMP_COMMIT_ID, Long.toString(maxUnsafeAutoIdTimestamp.get()));
                 commitData.put(HISTORY_UUID_KEY, historyUUID);
-                if (softDeleteEnabled) {
-                    commitData.put(Engine.MIN_RETAINED_SEQNO, Long.toString(softDeletesPolicy.getMinRetainedSeqNo()));
```

Comment:
how do we deal with the possible absence of this information on older commits?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/50415/commits/aa7e6ef271c602f5d19c31148e6e0a55088203b1 to set min_retained_seq_no to the max_seq_no of the safe commit. I also enabled the history assertion for non-soft-deletes indices.